crates/sui-core/src/consensus_validator.rs

```diff
@@ -57,6 +59,10 @@ impl TransactionValidator for SuiTxValidator {
         for tx in txs.into_iter() {
             match tx.kind {
                 ConsensusTransactionKind::UserTransaction(certificate) => {
+                    if certificate.epoch() != self.epoch_store.epoch() {
```

Comment:
This is not the best place to check this.
We should check it in add_to_verification_obligation? (we do so for the impl of the quorum signature info, but somehow not for the single auth sign info)

Reply:
Oh, so this particular place is a quorum sign info, so you are right, this check is already done as part of signature verification. Thanks for the pointer!

Reply:
https://github.com/MystenLabs/sui/pull/7478 to add verification to AuthoritySignInfo